addons/website_slides/models/slide_channel.py

```diff
@@ -641,7 +641,7 @@ def _resequence_slides(self, slide, force_category=False):
         index_of_added_slide = ids_to_resequence.index(slide.id)
         next_category_id = None
         if self.slide_category_ids:
-            force_category_id = force_category.id if force_category else slide.category_id.id
+            force_category_id = force_category or slide.category_id.id if slide.category_id else False
```

Comment:
Suggestion:
`category_id = force_category_id or slide.category_id.id`
(and then use category_id instead of force_category_id).

Easier to read.

Reply:
Thanks again for the advises.
1. If slide has no category_id, does `slide.category_id.id` will not trigger an error because accessing id on 'none'?
2. For the category changes, I had an error because the function is waiting for a category_id (integer) but in the test, a category instance is send so I replaced it to send only the id and so also refactoring the current function.

Reply:
`slide.category_id` will return an empty recordset if the slide has no category (and that's the case for all m2o actually).
Accessing any property on an empty recordset returns False (or None? Not sure bit they're both Falsy anyway).

So it should always work.

For the category change, which test was it related to?
If it's not the same test as the "new course" to "course" change, you could split the commit as explained in the review message.

Hope that helps.

Reply:
It helped, like always ;)
The related test is in the file `website_slides/tests/test_slide_utils.py`
I did a 2nd commit for the category changes.
Thanks again.